tools/ts-generator/types/functions.ts

```diff
@@ -246,8 +280,8 @@ const readArray = (readSize?: number) => <T>(
   obj?: FromBytes<T>
 ): [T[], number] => {
   const array: T[] = [];
-  const arraySize = readSize || buffer.readInt32LE(offset);
-  if (!readSize) {
+  const arraySize = readSize > -1 ? readSize : buffer.readInt32LE(offset);
```

Comment:
you can combine these two lines under one if statement so you don't perform two of the same conditional checks

Reply:
done, my friend